sirepo/package_data/static/js/raydata.js

```diff
@@ -547,6 +586,27 @@ SIREPO.app.directive('scansTable', function() {
                 $scope.reverseSortScans = ! $scope.reverseSortScans;
             };
 
+            $scope.togglePdfSelectAll = () => {
+                $scope.pdfSelectedScans = [];
+                if (! $scope.pdfSelectAllScans) {
```

Comment:
This logic took me a second to understand (and I'm still not sure I have it right). But I think that line 598 should be moved to the top and then the `!` can be removed here. I think that will make it clearer that in the case where pdfSelectAllScans is true we want to add all scans to pdfSelectedScans.

Reply:
34d5c6d1b0aa5be7fb841e47a6fbf0fc34ecd9e9